my-app/src/components/GenericForm.tsx

```diff
@@ -0,0 +1,147 @@
+}
+
+interface Props {
+  fields: Field[];
+  customSubmitFunction: (data: Record<string, any>) => void;
+  submitButtonName: string;
+  schema: Schema;
+}
+
+const GenericForm: React.FC<Props> = ({
+  fields,
+  customSubmitFunction,
+  submitButtonName,
+  schema,
+}) => {
+  const [formData, setFormData] = useState<Record<string, string | undefined>>(
+    {}
+  );
+  const [fieldsData, setFieldsData] = useState(fields);
+  const [currentLabelStyle, setCurrentLabelStyle] =
+    useState("signinLabelNormal");
+  const validate = ajv.compile(schema);
+
+  const {
+    register,
+    handleSubmit,
+    setValue,
+    formState: { errors },
+  } = useForm();
+
+  const handleInputChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const { id, value } = event.target;
+    setFormData({ ...formData, [id]: value });
+  };
+
+  const onChange = () => {
+    setCurrentLabelStyle("signinLabelNormal");
+  };
+  const validateForm = (data: Record<string, any>) => {
+    fields.map((field) => (field.errorMsgVisibility = true)); //Reset Error Messages
+    validate(data);
+    const errorMessages = Array.isArray(validate.errors) ? validate.errors : []; //Get all the errors
+    fields.map((field /*Go throu every field*/) =>
+      errorMessages.map(
+        (
+          message //Go throu every error
+        ) =>
+          message.instancePath.substring(1) == field.id //check if it is assign to the field id
+            ? (field.errorMsgVisibility = false) //if they match, show the error
+            : console.log()
+      )
+    );
+    setFieldsData(fields);
+  };
+
+  const onSubmit = async (data: Record<string, any>) => {
+    validateForm(data);
+    customSubmitFunction(data);
+  };
+
+  return (
+    <Box
+      component="form"
+      onSubmit={handleSubmit(onSubmit)}
+      sx={{ mt: 1 }}
+      onChange={onChange}
+    >
+      {fieldsData.map((field) => (
+        <Box key={field.id}>
+          <h4>{field.label}</h4>
+          {field.type !== "select" ? (
+          <input
+            type={field.type}
+            id={field.id}
+            {...register(field.id)}
+            required={field.required}
+            onChange={handleInputChange}
+            placeholder={
+              currentLabelStyle === "signinLabelNormal"
+                ? field.placeholder
+                : "Incorrect Credentials"
+            }
+            style={{
+              borderColor:
+                currentLabelStyle === "signinLabelNormal"
+                  ? "#181818"
+                  : "#d10000",
+            }}
+          />
+          ):(
+          <select
+          id={field.id}
+          {...register(field.id)}
+          required={field.required}
+          style={{
+            borderColor:
+              currentLabelStyle === "signinLabelNormal"
+                ? "#181818"
+                : "#d10000",
+          }}
+          >
+            {field.options?.map((option) => (
```

Comment:
Please break lines, create spaces, I find it really hard to read .
Example: 
![image](https://github.com/HelloIAmBanana/bank-hafifa/assets/33515941/6a1fa907-f366-4de3-9847-9aa4c0fe68f1)